src/daft-schema/src/dtype.rs

```diff
@@ -638,7 +649,29 @@ impl From<&ArrowType> for DataType {
             ArrowType::FixedSizeList(field, size) => {
                 Self::FixedSizeList(Box::new(field.as_ref().data_type().into()), *size)
             }
-            ArrowType::Map(field, ..) => Self::Map(Box::new(field.as_ref().data_type().into())),
+            ArrowType::Map(field, ..) => {
+                // todo: TryFrom in future? want in second pass maybe
```

Comment:
Are we doing this `TryForm`? Fine if we don't

Reply:
@jaychia and I want to do a TryFrom since it is fallible

Reply:
https://github.com/Eventual-Inc/Daft/issues/3015